### PR TITLE
Upgrade Typography to 0.15.x

### PR DESCRIPTION
**Pull request by KyleAMathews**
Hey! Love the look of your blog. And was really pleased to see that you're using Typography.js. Curious what you think about it?

I noticed it was a few versions behind so updated it.

I also played a bit with the styles which you can totally disregard but I generally like the body line height a bit lower so did that plus set the header line height to 1.1 (need to add this as an option to Typography.js as header line-height should almost always be different than the body's). Also bumped the body font size up a bit on desktop and down on mobile. Also added the Typography code plugin as it has a number of nice defaults for code but again YMMV.

Cheers!

### CODE REVIEW

**Comment by benmccormick** [comment]
Hey @KyleAMathews thanks so much for the pull request!  I really appreciate you taking the time to look into this and help make my blog better. (And I also am deeply grateful for the work you've put into Gatsby and Typography.js in general, I've loved using them to build my blog)

I had actually attempted to upgrade Typography.js earlier, and the issue I was hitting (which looks like it still exists in your branch) is that the upgrade played havoc on the spacing and sizing of my headers

image

Both the logo and the article/page titles to varying degrees got much bigger than I wanted them to be, and I wasn't immediately sure how to fix that, so I'd punted it off till I had more time.  I don't want to impose on your time, since I already really appreciate the PR, but I'd be much more willing to get this in if you had suggestions for fixing that.  I'm just not knowlegeable enough about the library to do it quickly, so it'll probably sit on the backburner for a little while otherwise.

**Comment by benmccormick** [comment]
Oh, and I do like the new line spacing btw, thank you for that improvement

File: utils/typography.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ const typography = new Typography({
     'Verdana',
     'sans-serif'
   ],
-  scale: 1.5,
+  scaleRatio: 1.5,
```
Comment (benmccormick) [inline utils/typography.js]: Ah that looks much better.  Only issue now is the spacing under the logo, which definitely is within my capabilities to figure out.  I'm done for the night right now, but will get that in and merged over the weekend.  Thanks so much!

**Comment by KyleAMathews** [comment]
Oh ooops! I'd forgotten that I'd changed another name, `scale` became `scaleRatio`. 

Fixed that in https://github.com/benmccormick/benmccormickorg/pull/1/commits/28007c5d36d368372cd6cd117f2ec1d3341f8dcb

You'd had it at 1.5 and the default (which it reverted back to with the push) is 2 hence the headers got much bigger. How this works is that the size of headers are based on a scale created from the `baseFontSize` and `scaleRatio`. The h1 is always one scale away from the`baseFontSize` so If the `baseFontSize` is 18px and `scaleRatio` is 2 then the h1 font size is 36px. All sizes should be based on this scale (that's what the scale function is for) so that you can adjust the `scaleRatio` to move things around.

File: utils/typography.js (hunk 1 above)
Comment (KyleAMathews) [inline utils/typography.js]: Cool! Always fun to see things being used on in the wild :-) Need to do another push to incorporate some more typography things I've learned into the library and get it to a 1.0.

**Comment by benmccormick** [comment]
@KyleAMathews Merged.  Thanks so much for the help!

**Pull request merged**